packages/workflow/src/internals.ts

```diff
@@ -315,9 +374,9 @@ async function handleWorkflowFailure(error: any) {
   state.completed = true;
 }
 
-async function completeQuery(queryId: string, result: unknown) {
+function completeQuery(queryId: string, result: unknown) {
   state.commands.push({
-    respondToQuery: { queryId, succeeded: { response: await state.dataConverter.toPayload(result) } },
+    respondToQuery: { queryId, succeeded: { response: state.dataConverter.toPayloadSync(result) } },
```

Comment:
Why'd all these need to become sync? Just curious

Reply:
I found out that I couldn't immediately cancel the workflow here:
https://github.com/temporalio/sdk-node/blob/1fea8fa89bb80de70f044644b6adc09f0072db77/packages/test/src/workflows/child-workflow-cancel.ts#L16

With async data converter instead of start and cancel reaching core, the scope is cancelled before the child schedule command is even sent.